ui/command_tab_view.cpp

```diff
@@ -98,9 +98,11 @@ void CommandTabView::OnSelectionChanged(const QModelIndex &index)
 {
     m_command_buffer_model->OnSelectionChanged(index);
 
-    // After m_command_buffer_view is filled out in CommandBufferModel::OnSelectionChanged(), expand
-    // the tree
-    m_command_buffer_view->expandAll();
+    // After m_command_buffer_view is filled out in CommandBufferModel::OnSelectionChanged(), do NOT
+    // expandAll. For huge trees (e.g. 20+ million nodes), it needs to traverse all expanded nodes
+    // to figure out how to setup scrollbar. So the more collapsed it is, the less it has to
+    // traverse
+    // m_command_buffer_view->expandAll();
```

Comment:
shall we remove this block completely?

Reply:
I kept it there as a warning in case we're ever tempted again to expandAll for UX reasons.
But if you prefer removing that, I can